app/uk/gov/hmrc/transitmovements/services/XmlParsers.scala

```diff
@@ -48,4 +50,20 @@ object XmlParsers extends XmlParsingServiceHelpers {
       case exception: DateTimeParseException => Left(ParseError.BadDateTime("preparationDateAndTime", exception))
     }
 
+  val messageTypeExtractor: Flow[ParseEvent, ParseResult[MessageType], NotUsed] = Flow[ParseEvent]
+    .statefulMapConcat {
+      () =>
+        _ match {
+          case s: StartElement if MessageType.values.exists(_.rootNode == s.localName) =>
+            Seq(MessageType.values.find(_.rootNode == s.localName).get)
+          case _ => Seq.empty
+        }
+    }
+    .prefixAndTail(1)
```

Comment:
We're not using the resulting stream, so can we just use `take(1)` to let Akka complete the stream? Otherwise, we might end up with a memory leak with a stream that hasn't completed.